reference/method-patterns.md

```diff
@@ -4,54 +4,87 @@ description: A simple and powerful way of identifying method definitions and inv
+| `java.lang.String *(int)`                   | Any method on `String` that accepts a single argument of type `int`.                                                                                                                                             |
+| `java.lang.String format(String, ..)`       | Exactly the [String.format(String format, Object... args)](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#format(java.lang.String,java.lang.Object...)) method with varargs. |
+| `java.lang.String format(String, Object[])` | Exactly the [String.format(String format, Object... args)](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#format(java.lang.String,java.lang.Object...)) method with varargs. |
+| `com.yourorg.Foo bar(int, String, ..)`      | Any method on `Foo` named `bar` accepting an `int`, a `String`, and zero or more other arguments of any type.                                                                                                    |
+| `com.yourorg.Foo <constructor>(*, *, *)`    | Constructors of `Foo` accepting exactly three arguments of any type.                                                                                                                                             |
+| `*..String *(..)`                           | Any method accepting any arguments on classes named "String" in any package.                                                                                                                                     |
+| `*..* *(..)`                                | Any method accepting any arguments on any class.                                                                                                                                                                 |
+| `*..* foo(..)       `                       | Any method named `foo` which accepts any number of arguments. This is the equivalent of doing a generic text search for methods named `foo`.                                                                     |
+| `org..* foo(..)`                            | Any method named `foo` in a package that starts with `org`. This could be `orgmeow` or it could be `org.meow.bark` or so on.                                                                                     |
+| `org.openrewrite.java.* foo(..)`            | Any method named `foo` in a package that starts with `org.openrewrite.java.`. Note that the `.*` after `java` means that the method _must_ be in this package rather than some subsequent subpackages.           |
+| `org.Foo bar(java.util.List)`               | Exactly the `bar` method in the `org.Foo` package that takes in a single `java.util.List` as an argument.                                                                                                        |
```

Comment:
With some of these I was wondering if this is actually how things work, such as the case matching `orgmeow`; did you find or test these somewhere? In the past we've had issues where our [docs were a little aspirational](https://rewriteoss.slack.com/archives/C01A843MWG5/p1686582393410069?thread_ts=1686563285.282649&cid=C01A843MWG5) rather than reflecting current afairs, and I'd like to avoid adding to the confusion.

Reply:
Jonathan wrote out examples here: https://moderneinc.slack.com/archives/C01S4M53THT/p1690921625049179?thread_ts=1690917553.347359&cid=C01S4M53THT

I was largely just taking his examples and converting them to this. It's possible that these don't work, though (as I never tested any of these).

I'll see if I can figure out how to test these to confirm.

Reply:
@timtebeek FYI I can confirm that the `orgmeow` thing does indeed not work. I updated the docs to reflect that. `org..*` would match `org.meow` or `org.springframework` but _not_ `orgmeow`. I'm not entirely sure why but 🤷